# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLBuffer.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLBuffer.h
--- a/dom/canvas/WebGLBuffer.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLBuffer.h	Wed Jan 08 22:19:16 2020 +0000
@@ -10,19 +10,19 @@
 
 #include "CacheInvalidator.h"
 #include "GLDefs.h"
-#include "mozilla/LinkedList.h"
-#include "nsWrapperCache.h"
 #include "WebGLObjectModel.h"
 #include "WebGLTypes.h"
 
 namespace mozilla {
 
-class WebGLBuffer final : public WebGLRefCountedObject<WebGLBuffer>,
-                          public LinkedListElement<WebGLBuffer> {
+class WebGLBuffer final : public WebGLContextBoundObject {
   friend class WebGLContext;
   friend class WebGL2Context;
+  friend class WebGLMemoryTracker;
   friend class WebGLTexture;
 
+  MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(WebGLBuffer, override)
+
  public:
   enum class Kind { Undefined, ElementArray, OtherData };
 
@@ -31,8 +31,6 @@
   void SetContentAfterBind(GLenum target);
   Kind Content() const { return mContent; }
 
-  void Delete();
-
   size_t SizeOfIncludingThis(mozilla::MallocSizeOf mallocSizeOf) const;
 
   GLenum Usage() const { return mUsage; }
@@ -51,10 +49,8 @@
 
   const GLenum mGLName;
 
-  NS_INLINE_DECL_REFCOUNTING(WebGLBuffer)
-
  protected:
-  ~WebGLBuffer();
+  ~WebGLBuffer() override;
 
   void InvalidateCacheRange(uint64_t byteOffset, uint64_t byteLength) const;
 